lib/rspec/matchers/built_in/respond_to.rb

```diff
@@ -125,7 +125,14 @@ def matches_arity?(actual, name)
 
           return true if expectation.empty?
 
-          signature = Support::MethodSignature.new(Support.method_handle_for(actual, name))
+          method_handle = Support.method_handle_for(actual, name)
+
+          signature =
```

Comment:
CI is broken on linting

```
lib/rspec/matchers/built_in/respond_to.rb:112:9:C: Metrics/MethodLength: Method has too many lines. [19/15]
        def matches_arity?(actual, name)...
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
48 files inspected1 offense detected
```

Maybe `signature` assignment can be extracted as method?